src/helpers/time-to-first-byte-analyser.helper.js

```diff
@@ -0,0 +1,19 @@
+import JSONPerformanceReportParser from './JSON-performance-report-parser.helper';
+
+class TimeToFirstByteAnalyser {
+  constructor(jsonPerformanceReportParser) {
+    this.reader = jsonPerformanceReportParser;
+  }
+
+  analyseTTFB(parsedReport, maxTiming) {
+    return parsedReport.filter(parsedReport => parsedReport.ttfb > maxTiming);
+  }
+
+  checkTiming(fileName, maxTiming) {
+    const parsedReport = this.reader.parse(fileName);
+
+    return this.analyseTTFB(parsedReport, maxTiming);
+  }
+}
+
+export const analyser = new TimeToFirstByteAnalyser(new JSONPerformanceReportParser());
```

Comment:
This is not what I'm looking for. It's much harder to test right now. You cannot mock parser result so you cannot test all of the possibilities here.

`export const create = (reportParser = new JSONPerformanceReportParser()) => new TimeToFirstByteAnalyser(reportParser);`

This is the correct way. You might ask why - right now I'm able to run it like this

`
import { create as createAnalyser) from './time-to-first-byte-analyser.helper';
const analyser = createAnalyser();
`

but I'm also able to do so:


`
import { create as createAnalyser) from './time-to-first-byte-analyser.helper';

const myFakeParser = {
  parse: () => {
    return [
       {ttfb:0 url:'some-url'},
    ]
  }
}

const analyser = createAnalyser();
`

Do you see a difference ?